File: Algorithms/Easy/283.move_zeros.js
const moveZeros = (nums) => {
  let numOfZeros = 0;

  for (let i = 0; i < nums.length; i++) {
    if (nums[i] == 0) numOfZeros++;
    else if (numOfZeros > 0) {
      let temp = nums[i];
      nums[i] = 0;
      nums[i - numOfZeros] = temp;
    }
  }
  console.log(nums);
};

moveZeros([0, 1, 0, 3, 12]);

File: Algorithms/Easy/283.move_zeros.py
def move_zeros(nums:list) -> None:
    slow = 0

    for fast in range(len(nums)):
        if nums[slow] == 0 and nums[fast] !=0:
            nums[fast], nums[slow] = nums[slow], nums[fast]

        if nums[slow] != 0:
            slow += 1
    print(nums)

move_zeros([0,1,0, 3, 12]) 
